pkg/providers/gcp/provider_postgres.go

```diff
@@ -472,7 +498,30 @@ func (p *PostgresProvider) buildCloudSQLCreateStrategy(ctx context.Context, pg *
 	}
 
 	if instance.Settings == nil {
-		instance.Settings = &gcpiface.Settings{}
+		instance.Settings = &gcpiface.Settings{
+			Tier:                   defaultTier,
+			AvailabilityType:       defaultAvailabilityType,
+			StorageAutoResizeLimit: defaultStorageAutoResizeLimit,
+			StorageAutoResize:      utils.Bool(defaultStorageAutoResize),
+			BackupConfiguration: &gcpiface.BackupConfiguration{
+				Enabled:                    utils.Bool(defaultBackupConfigEnabled),
+				PointInTimeRecoveryEnabled: utils.Bool(defaultPointInTimeRecoveryEnabled),
+				BackupRetentionSettings: &gcpiface.BackupRetentionSettings{
+					RetentionUnit:   defaultBackupRetentionSettingsRetentionUnit,
+					RetainedBackups: defaultBackupRetentionSettingsRetainedBackups,
+				},
+			},
+			DataDiskSizeGb:            defaultDataDiskSizeGb,
+			DeletionProtectionEnabled: utils.Bool(defaultDeleteProtectionEnabled),
+			IpConfiguration: &gcpiface.IpConfiguration{
+				Ipv4Enabled: utils.Bool(defaultIPConfigIPV4Enabled),
+			},
+			UserLabels: tags,
```

Comment:
I think this is a rebase error, setting all of the defaults in the initial struct was removed in a recent PR. Instead you can check that the `DenyMaintenancePeriods != nil` and set it there.

Reply:
Thank you! Done